src/test/java/net/minecraftforge/debug/world/BiomeTemperatureModifierTest.java

```diff
@@ -0,0 +1,90 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+package net.minecraftforge.debug.world;
+
+import net.minecraft.core.Registry;
+import net.minecraft.data.worldgen.SurfaceBuilders;
+import net.minecraft.resources.ResourceKey;
+import net.minecraft.resources.ResourceLocation;
+import net.minecraft.world.level.biome.AmbientMoodSettings;
+import net.minecraft.world.level.biome.Biome;
+import net.minecraft.world.level.biome.Biome.BiomeCategory;
+import net.minecraft.world.level.biome.Biome.Precipitation;
+import net.minecraft.world.level.biome.BiomeGenerationSettings;
+import net.minecraft.world.level.biome.BiomeSpecialEffects;
+import net.minecraft.world.level.biome.MobSpawnSettings;
+import net.minecraftforge.common.BiomeDictionary;
+import net.minecraftforge.common.BiomeManager;
+import net.minecraftforge.common.BiomeManager.BiomeEntry;
+import net.minecraftforge.fml.common.Mod;
+import net.minecraftforge.fml.event.lifecycle.FMLCommonSetupEvent;
+import net.minecraftforge.fml.javafmlmod.FMLJavaModLoadingContext;
+import net.minecraftforge.fmllegacy.RegistryObject;
+import net.minecraftforge.registries.DeferredRegister;
+import net.minecraftforge.registries.ForgeRegistries;
+
+@Mod(BiomeTemperatureModifierTest.MOD_ID)
+public class BiomeTemperatureModifierTest 
+{
+	static final String MOD_ID = "biome_temperature_modifier_test";
+    private static final boolean ENABLED = false;
+    public static final DeferredRegister<Biome> BIOMES = DeferredRegister.create(ForgeRegistries.BIOMES, MOD_ID);
+    public static final ResourceKey<Biome> TEST_BIOME_KEY = ResourceKey.create(Registry.BIOME_REGISTRY, new ResourceLocation(MOD_ID, "test_biome"));
+    public static final RegistryObject<Biome> TEST_BIOME = BIOMES.register("test_biome", () -> BiomeTemperatureModifierTest.TEST_BIOME_BIOME);
+
+    public BiomeTemperatureModifierTest() 
+    {
+        if (!ENABLED) return;
+        BIOMES.register(FMLJavaModLoadingContext.get().getModEventBus());
+        FMLJavaModLoadingContext.get().getModEventBus().addListener(this::commonSetup);
+    }
+
+    private void commonSetup(FMLCommonSetupEvent event) 
+    {
+        event.enqueueWork(() -> {
+            BiomeDictionary.addTypes(TEST_BIOME_KEY, BiomeDictionary.Type.MODIFIED);
+            BiomeManager.addBiome(BiomeManager.BiomeType.WARM, new BiomeEntry(TEST_BIOME_KEY, 100));
+        });
+    }
+
+    /*
+     * the Biome is only made for testing, the parameters which are used are test parameters
+     */
+	private static final Biome TEST_BIOME_BIOME = new Biome.BiomeBuilder()
```

Comment:
This should be registered properly, not as a static initialized entry.

Reply:
t/fill

Reply:
i will move this directly into the Supplier of the RegistryObject,
and about the json Biome, is it possible to generate it via the Data generator?
and if yes, where do i need to set the mod id for the generator?

Reply:
Currently there is no data generator for world gen objects, there is a PR for it though: #7257

Reply:
okay good to know